nbgrader/install.py

```diff
@@ -0,0 +1,176 @@
+from IPython.utils.py3compat import cast_unicode_py2
+
+
+def _get_profile_dir(profile, ipython_dir):
+    if not ipython_dir:
+        pdir = ProfileDir.find_profile_dir_by_name(get_ipython_dir(), profile).location
+    else:
+        ipython_dir = os.path.expanduser(ipython_dir)
+        pdir = ProfileDir.find_profile_dir_by_name(ipython_dir, profile).location
+    return pdir
+
+def install(profile='default', symlink=True, user=False, prefix=None,
+            verbose=False, ipython_dir=None):
+    """Install and activate nbgrader on a profile
+    """
+
+    dname = os.path.dirname(__file__)
+    # might want to check if already installed and overwrite if exist
+    if symlink and verbose:
+        print('I will try to symlink the extension instead of copying files')
+    if prefix and verbose:
+        print("I will install in prefix:", prefix)
+    if not ipython_dir:
+        nbextensions_dir = None
+    else:
+        nbextensions_dir = os.path.join(ipython_dir,'nbextensions')
+
+    nbe.install_nbextension(os.path.join(dname, 'nbextensions', 'nbgrader'),
+                            user=user,
+                            prefix=prefix,
+                            symlink=symlink,
+                            nbextensions_dir=nbextensions_dir
+                            )
+
+
+# TODO pass prefix as argument.
+def activate(profile=None, ipython_dir=None):
+    """
+    Manually modify the frontend json-config to load nbgrader extension.
+    """
+
+    pdir = _get_profile_dir(profile, ipython_dir)
+    json_dir = os.path.join(pdir, 'nbconfig')
+    json_file = os.path.expanduser(os.path.join(json_dir, 'notebook.json'))
+
+    try:
+        with io.open(json_file, 'r') as f:
+            config = json.loads(f.read())
+    except IOError:
+        # file doesn't exist yet. IPython might have never been launched.
+        config = {}
+
+    if not config.get('load_extensions', None):
+        config['load_extensions'] = {}
+    config['load_extensions']['nbgrader/create_assignment'] = True
+
+    if not os.path.exists(json_dir):
+        os.mkdir(json_dir)
+
+    with io.open(json_file, 'w+') as f:
+        f.write(cast_unicode_py2(json.dumps(config, indent=2), 'utf-8'))
+
+
+def deactivate(profile=None, ipython_dir=None):
+    """
+    Manually modify the frontend json-config to load nbgrader extension.
+    """
+
+    pdir = _get_profile_dir(profile, ipython_dir)
+    json_dir = os.path.join(pdir, 'nbconfig')
+    json_file = os.path.expanduser(os.path.join(json_dir, 'notebook.json'))
+
+    try:
+        with io.open(json_file, 'r') as f:
+            config = json.loads(f.read())
+    except IOError:
+        # file doesn't exist yet. IPython might have never been launched.
+        return
+
+    if 'load_extensions' not in config:
+        return
+    if 'nbgrader/create_assignment' not in config['load_extensions']:
+        return
+
+    config['load_extensions']['nbgrader/create_assignment'] = False
```

Comment:
I don't think this will deactivate the ext. Actually the JS config use a dict as a set, but set do not exist in json, the value dont' have effet yet. You we might have to delete the key. 
Let me check.